adapters/au_vic.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+import { REQUEST_TIMEOUT } from '../lib/constants';
+import { default as baseRequest } from 'request';
+import _ from 'lodash';
+import { default as moment } from 'moment-timezone';
+import parallelLimit from 'async/parallelLimit';
+import Bottleneck from 'bottleneck';
+const request = baseRequest.defaults({timeout: REQUEST_TIMEOUT});
+
+// Default rate limiting on API is set to 5 requests/sec.
+// > Please send an email to Tools.support@epa.vic.gov.au with subject
+// > 'EPA API Access Request: Increase rate-limiting' and a justified
+// > reason if you want to get it increased for your subscriptions.
+var limiter = new Bottleneck({
+  reservoir: 5, // allow 5 requests
+  reservoirRefreshAmount: 5,
+  reservoirRefreshInterval: 1000, // every 1000ms
+  maxConcurrent: 1,
+  minTime: 200 + 50 // to stagger requests out throught each second (1000 / 5) adding a 50ms buffer
+});
+
+export const name = 'victoria';
+
+export function fetchData (source, cb) {
+  limiter.submit(request, {
+    url: source.url,
+    headers: {
+      'X-API-Key': process.env.EPA_VIC_TOKEN
+    }
+  }, function (err, res, body) {
+    if (err || res.statusCode !== 200) {
+      return cb({message: 'Failure to load data url.'});
+    }
+
+    // Wrap everything in a try/catch in case something goes wrong
+    try {
+      // Format the data
+      formatData(body, function (err, data) {
+        // Make sure the data is valid
+        if (err || data === undefined) {
+          return cb({message: 'Failure to parse data.'});
+        }
+        cb(null, data);
+      });
+    } catch (e) {
+      return cb({message: 'Unknown adapter error.'});
+    }
+  });
+}
+
+var parameters = {
+  'PM2.5': 'pm25',
+  'PM10': 'pm10',
+  'NO2': 'no2',
+  'SO2': 'so2',
+  'O3': 'o3',
+  'CO': 'co',
+  'BC': 'bc'
+};
+
+var units = {
+  '&micro;g/m&sup3;': 'µg/m³',
+  'ppm': 'ppm'
+};
+
+var formatData = function (data, formatDataCB) {
+  var sites = JSON.parse(data).records;
+
+  // request measurements from each site
+  var tasks = sites.map(function (site) {
+    return function (cb) {
+      limiter.submit(request, {
+        url: `https://gateway.api.epa.vic.gov.au/environmentMonitoring/v1/sites/${site.siteID}/parameters`,
+        headers: {
+          'X-API-Key': process.env.EPA_VIC_TOKEN
+        }
+      }, function (err, res, body) {
+        if (err || res.statusCode !== 200) {
+          console.error(err, res);
+          return cb({message: 'Failure to load data url.'});
+        }
+        var source = JSON.parse(body);
+
+        // base properties shared for all measurements at this site
+        var baseProperties = {
+          location: source.siteName,
+          city: 'Victoria',
```

Comment:
It's going to be hard to set this, I'd rather not hardcode them all, but upstream provider doesn't provide a city. Victoria is a state not a city, ant not all the records are in Melbourne, and some not really in any city.

Reply:
I took another look, I've hardcode a mapping of site name to city, and anyones not hardcoded the site name is the city name.

Reply:
perfect, thank you!